#1.
Неверно были указаны "" в указании строки с ip адресом. Исправил.
```
{ "info" : "Sample JSON output from our service\t",
    "elements" :[
        { "name" : "first",
        "type" : "server",
        "ip" : 7175 
        },
        { "name" : "second",
        "type" : "proxy",
        "ip" : "71.78.22.43"
        }
    ]
}
```
#2
```
#!/usr/bin/env python3

import socket
import datetime
import json
import yaml
import time


i = 1
dom = {'drive.google.com':'0.0.0.0', 'mail.google.com':'0.0.0.0', 'google.com':'0.0.0.0'}
while 1 == 1:
  for host in dom:
    ip = socket.gethostbyname(host)
    if ip != dom[host]:
        print(str(datetime.datetime.now().strftime("%Y-%m-%d %H:%M:%S")) +' [ERROR] ' + str(host) +' IP mistmatch: '+dom[host]+' '+ip)
        dom[host] = ip

    with open("C:/Users/Lidron/Desktop/devops-netology/ser_" + host + ".json", 'w') as js:
        json_data = json.dumps([{host: ip}])
        js.write(json_data)
    with open("C:/Users/Lidron/Desktop/devops-netology/ser_" + host + ".yml", 'w') as ym:
        yml_data = yaml.dump([{host: ip}])
        ym.write(yml_data)
  i += 1
  time.sleep(10)
```
##Вывод: 
```
C:\Users\Lidron\Desktop\devops-netology>python 4.3.2.py
2021-09-27 23:51:56 [ERROR] drive.google.com IP mistmatch: 0.0.0.0 209.85.233.194
2021-09-27 23:51:56 [ERROR] mail.google.com IP mistmatch: 0.0.0.0 108.177.14.83
2021-09-27 23:51:56 [ERROR] google.com IP mistmatch: 0.0.0.0 173.194.221.139
```
Так же создались файлы для каждого домена, в json и yml.